torch/ao/sparsity/experimental/data_sparsifier/lightning/callbacks/data_sparsity.py

```diff
@@ -0,0 +1,59 @@
+from copy import deepcopy
+import torch
+from dataclasses import dataclass, field
+from typing import Any, Optional, Dict
+import pytorch_lightning as pl
+
+from ._data_sparstity_utils import _attach_model_to_data_sparsifier, _log_sparsified_level
+
+
+@dataclass
+class _DataSparsity(pl.callbacks.Callback):
+    data_sparsifier_type: Any
+    data_sparsifier_args: Dict  # this the arguments for sparsifier_type (except data_list)
+    data_sparsifier: Any = field(init=False, default=None)
+    sparsified: Optional[torch.nn.Module] = field(init=False, default=None)
+
+    def __post_init__(self):
+        pass
+
+
+@dataclass
+class PostTrainingDataSparsity(_DataSparsity):
+    """Lightning callback that enables post-training sparsity.
+
+    This callback aims to sparsify the model inside lightning module after training.
+    **Note that the model is copied and then sparsified, so the existing model is not modified**
+
+    The sparsified model can be used for comparison and can be accessed using
+        <callback_obj>.sparsified
+
+    Args:
+        data_sparsifier_type (some implemented class of BaseDataSparsifier)
```

Comment:
maybe just `data_sparsifier_class`? I feel type is a bit ambiguous

Reply:
Yeah, makes sense. Changed it to data_sparsifier_class for now.